cli/cli/gateway.py

```diff
@@ -145,14 +168,22 @@ def get_routes(self) -> list[Route]:
             service_protocol = service["protocol"]
             service_url = f"{service_protocol}://{service_host}:{service_port}"
 
-            routes.append(
-                Route(
-                    relative_url=route_path,
-                    target=service_url,
-                    http_methods=http_methods,
-                )
+            r = Route(
+                relative_url=route_path,
+                target=service_url,
+                http_methods=http_methods,
             )
 
+            # Check if route has plugins installed
+            if route_id in route_plugins:
+                for p in route_plugins[route_id]:
```

Comment:
I don't really like the quadruple nesting level, we can remove one by doing this:

```suggestion
            for p in route_plugins.get(route_id) or []:
```